### ISSUE TITLE
takeLatest/takeEvery implementing while(true) can be confusing for people learning Saga

### ISSUE THREAD

**Issue by Zacqary** [yelouafi/redux-saga#677]
I've been ramping up some new developers on Redux-Saga, and one of the key understandings is that sagas that watch for actions generally require an infinite loop:
```javascript
while(true) {
   const action = yield take(SOME_ACTION);
}
```

But then I started seeing in code review things like this:
```javascript
while(true) {
   yield takeLatest(SOME_ACTION, actionHandler);
}
```

Now, obviously this could have been avoided if:
- I had encouraged people to learn `takeEvery` more; I've gotten out of the habit of using it since we use `Symbol`s for our action types and `takeEvery` didn't support that (has that bug been fixed?)
- My devs had read the docs for `takeLatest` more carefully and realized it already implements the `while(true)` for you

Still, though, I think this inconsistency is a pain point. I wonder if there's a better solution.

**Comment by Andarist** [comment]
Could you give examples of your symbol usage?

About mentioned inconsistency - is it rly one? Its a well documented API
and im not sure what could be done to improve the situation.

Also - keep an eye on takeEvery/takeLatest changes in PRs. They are going
to be effects as well, and the traditional import will be deprecated soon
as an attempt to soften a confusion between effects and helpers.

**Comment by Zacqary** [comment]
Instead of `const ACTION_TYPE = "ACTION_TYPE"` our convention is to do `const ACTION_TYPE = Symbol("LocalNamespace ACTION_TYPE")`. Turns out this was fixed in #246, so `takeEvery` should be safe for us to use now and I need to get back in that habit.

Changing `takeEvery`/`takeLatest` to effects would be a great help. But furthermore, I guess the reason I see confusion is because they're called `take[Something]` but their behavior is actually much closer to `fork` than to `take`.

I do recognize that the way the docs are structured is to introduce `takeEvery` first and then consider `take` to be an advanced concept, so again, maybe it's my fault for teaching people Saga in the opposite direction. Although I do think it's valuable to understand what's actually going on under the hood in a saga — seeing an infinite loop of `take`s is a very good way to communicate the concept of why Saga uses generators.

In fact, to me that's the power of Saga over something with a huge library like RxJS — the fact that you can express complex patterns with these simple building blocks. I agree, I'm not sure off the top of my head how to do it better, but I would like it if the higher-level helpers were more in harmony with the lower-level building blocks that they're named after.

**Comment by Andarist** [comment]
Hm. I see your point. However those names are really descriptive about what
they do and thats huge pro here. Naming them after fork would for sure
introduce much bigger mess
